### PR TITLE
Bugfix: ember emitter, cleaner and squishy block (get valid sector)

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**

### CODE REVIEW

File: TombEngine/Game/misc.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ void TargetNearestEntity(ItemInfo* item, CreatureInfo* creature)
 	}
 }
 
-bool IsNextSectorValid(const ItemInfo& item, const Vector3& dir, float dist)
+bool IsNextSectorValid(const ItemInfo& item, const Vector3& dir, float dist, float canFloat)
```
Comment (Sezzary) [inline TombEngine/Game/misc.cpp:43]: `canFloat` should be a bool.

File: TombEngine/Game/misc.cpp [hunk 2]
```diff
@@ -59,15 +59,22 @@ bool IsNextSectorValid(const ItemInfo& item, const Vector3& dir, float dist)
 	{
 		// Test for step.
 		int relFloorHeight = abs(pointColl.Position.Floor - item.Pose.Position.y);
-		if (relFloorHeight >= CLICK(1) && item.Pose.Position.y >= pointColl.Position.Floor)
-			return false;
+
+		if (relFloorHeight >= CLICK(1) && item.Pose.Position.y >= pointColl.Position.Floor && canFloat)
```
Comment (Sezzary) [inline TombEngine/Game/misc.cpp:63]: Remove double tabs on return lines. Block like this one can be formated as follows:

if (x)
{
    // do A
}
else if (y)
{
    // do B
}

File: TombEngine/Objects/Effects/EmberEmitter.cpp [hunk 3]
```diff
@@ -27,10 +27,14 @@ namespace TEN::Effects::EmberEmitter
 		
 		if (item.TriggerFlags < 0)
 		{
+
```
Comment (Sezzary) [inline TombEngine/Objects/Effects/EmberEmitter.cpp:30]: Remove newline here.

File: TombEngine/Game/misc.cpp (hunk 1 above)
Comment (Nemoel-Tomo) [reply TombEngine/Game/misc.cpp]: aah that was a bad mistake, yes it should be bool.  fixed^^

File: TombEngine/Game/misc.cpp (hunk 2 above)
Comment (Nemoel-Tomo) [reply TombEngine/Game/misc.cpp:63]: fixed

File: TombEngine/Objects/Effects/EmberEmitter.cpp (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Effects/EmberEmitter.cpp]: fixed

**Review by Sezzary — APPROVED** [review]

**Pull request merged**